internal/flusher/flusher.go

```diff
@@ -0,0 +1,45 @@
+package flusher
+
+import (
+	"github.com/ozoncp/ocp-requirement-api/internal/models"
+	"github.com/ozoncp/ocp-requirement-api/internal/repository"
+	"github.com/ozoncp/ocp-requirement-api/internal/utils"
+	"log"
+)
+
+type Flusher interface {
+	Flush(entities []models.Requirement) []models.Requirement
+}
+
+type flusher struct {
+	chunkSize  int
+	repository repository.Repo
+}
+
+func NewFlusher(chunkSize int, repository repository.Repo) Flusher {
+	return flusher{
+		chunkSize:  chunkSize,
+		repository: repository,
+	}
+}
+
+func (f flusher) Flush(requirements []models.Requirement) []models.Requirement {
+	requirementsBulks, err := utils.SplitRequirementsToBulks(requirements, f.chunkSize)
+
+	flushedRequirements := make([]models.Requirement, 0)
+
+	if err != nil {
+		log.Print(err)
+		return flushedRequirements
+	}
+
+	for _, requirements := range requirementsBulks {
+		if err := f.repository.AddEntities(requirements); err == nil {
+			flushedRequirements = append(flushedRequirements, requirements...)
+		} else {
+			log.Print(err)
+		}
```

Comment:
В Go есть негласная конвенция раннего выхода: то есть если есть возможность не использовать else, то его лучше не использовать. Тем самым уменьшив вложенность и улучшив читаемость (может быть). То есть можно написать вот так:
```suggestion
		if err := f.repository.AddEntities(requirements); err != nil {
			log.Print(err)
			continue
		}
		flushedRequirements = append(flushedRequirements, requirements...)
```
Это не обязательно, но стоит иметь в виду

Reply:
Ну это не только в го. Согласен